src/Executor/ReferenceExecutor.php

```diff
@@ -917,11 +917,11 @@ private function completeListValue(ListOfType $returnType, $fieldNodes, ResolveI
             'User Error: expected iterable, but did not find one for field ' . $info->parentType . '.' . $info->fieldName . '.'
         );
         $containsPromise = false;
-        $i               = 0;
+
         $completedItems  = [];
-        foreach ($result as $item) {
+        foreach ($result as $idx => $item) {
             $fieldPath     = $path;
-            $fieldPath[]   = $i++;
+            $fieldPath[]   = $idx;
```

Comment:
I'd rather not use abbreviations. `$index` would be 👌

Reply:
SGTM